parachain/src/primitives.rs

```diff
@@ -62,6 +62,9 @@ impl ValidationCode {
 	}
 }
 
+/// A hash of the parachain validation code.
+pub type ValidationCodeHash = Hash;
```

Comment:
I'd prefer the newtype pattern for more type safety, although it's more boilerplate as well.

Reply:
Agreed, I have the changes ready, but they're touching a lot of stuff, probably easier to review if I make them on a follow-up PR.